server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1212,6 +1225,10 @@ public OptionalLong getForecastedShardSizeInBytes() {
         return shardSizeInBytesForecast == null ? OptionalLong.empty() : OptionalLong.of(shardSizeInBytesForecast);
     }
 
+    public Map<String, Set<String>> getFieldsForModels() {
+        return fieldsForModels;
```

Comment:
Should we ensure that the map returned here is always immutable (or at least unmodifiable)?

Reply:
Good call. Let's use an `ImmutableOpenMap`, like it's done on the `aliases` field 👍 . We can adjust as necessary with other folks expertise here.

Reply:
Commit https://github.com/elastic/elasticsearch/pull/103319/commits/c5748f70ea4ff132567b5cb1b41a0402bd065b45 should ensure that `IndexMetadata.Builder` always passes an `ImmutableOpenMap` to the `IndexMetadata` constructor.